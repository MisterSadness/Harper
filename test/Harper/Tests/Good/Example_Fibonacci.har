fib :: Integer -> Integer;
fib n = {
  var (previous :: Integer) = 1;
  var (current :: Integer) = 1;
  var (i :: Integer) = 2;
  while i < n {
    (next :: Integer) = previous + current;
    previous := current;
    current := next;

    i += 1;
  }

  return current;
};

main :: Integer;
main = fib 42;