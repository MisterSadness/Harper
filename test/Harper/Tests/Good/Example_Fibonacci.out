Execution ended with value:
267914296